carsus/io/cmfgen/base.py

```diff
@@ -0,0 +1,440 @@
+import numpy as np
+import pandas as pd
+import itertools
+import gzip
+import warnings
+from carsus.io.base import BaseParser
+
+
+# TODO: add `skiprows` parameter
+def find_row(fname, string1, string2='', how='both', num_row=False):
+    """Search strings inside plain text files and returns matching\
+    line or row number.
+
+    Parameters
+    ----------
+    fname : str
+        Path to plain text file.
+    string1 : str
+        String to search.
+    string2 : str
+        Extra string to search (default is '').
+    how : {'one', 'both', 'first'}
+        If 'both' search for string1 AND string2. If 'one' search for string1\
+            OR string2. If 'first' searches for 'string1' AND NOT string2\
+            (default is 'both').
+    num_row : bool
+        If true, returns row number instead (default is False).
+
+    Returns
+    -------
+    str or int
+        Returns matching line or row number.
+
+    """
+    with open(fname, encoding='ISO-8859-1') as f:
```

Comment:
why that encoding and not `utf8`

Reply:
That's the first problem we encountered when writing parsers. Some files couldn't be read with UTF-8 (files with author name Galav**í**s) and that encoding worked for all. The failing files were:

AL/IX/23oct02/fin_osc
AL/VIII/23oct02/fin_osc
MG/VIII/23oct02/fin_osc